### PR TITLE
Feat: Introducing Markdown linting using Remark

### PR DESCRIPTION
**Pull request by literat**

### CODE REVIEW

File: packages/remark-config/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "type": "module",
+  "main": "index.js",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "remark-preset-lint-consistent": "^6.0.0",
+    "remark-preset-lint-markdown-style-guide": "^6.0.0",
+    "remark-preset-lint-recommended": "^7.0.0"
+  }
```
Comment (adamkudrna) [inline packages/remark-config/package.json:31]: Cool! 👍🏻 I'm looking forward to generating TOC using Remark 🎉.

https://github.com/remarkjs/remark-toc

File: packages/remark-config/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ['lint-maximum-line-length', Infinity],
+    // warn when emphasis markers violate the given style
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-emphasis-marker
+    ['lint-emphasis-marker', '_'],
+    // warn when table pipes are not aligned
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-table-pipe-alignment
+    ['lint-table-pipe-alignment', false],
+    // warn when table cells are incorrectly padded
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-table-cell-padding
+    ['lint-table-cell-padding', 'consistent'],
```
Comment (adamkudrna) [inline packages/remark-config/index.js:25]: Will the tables setup work with Prettier?

File: packages/remark-config/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    'preset-lint-recommended',
+    'preset-lint-consistent',
+    'preset-lint-markdown-style-guide',
+
+    // warn when the spacing between a list item’s bullet and its content violates a given style
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-list-item-indent
+    ['lint-list-item-indent', 'one'],
+    // warn when lines are too long
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-maximum-line-length
+    ['lint-maximum-line-length', Infinity],
```
Comment (adamkudrna) [inline packages/remark-config/index.js:16]: 🎉

File: packages/remark-config/index.js
Comment (adamkudrna) [inline packages/remark-config/index.js:1]: Good choices here, I believe 👍🏻

File: packages/remark-config/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    // warn when emphasis markers violate the given style
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-emphasis-marker
+    ['lint-emphasis-marker', '_'],
```
Comment (adamkudrna) [inline packages/remark-config/index.js:19]: 👍🏻

**Review by adamkudrna — APPROVED** [review]
Nice! I'm eager to start using it in Spirit 🙂.

File: packages/remark-config/package.json (hunk 1 above)
Comment (literat) [reply packages/remark-config/package.json:31]: Yeah, I saw this somewhere in another design system. Maybe IBM Carbon or Adobe Spectrum are generating their TOCs.

File: packages/remark-config/index.js (hunk 2 above)
Comment (literat) [reply packages/remark-config/index.js:25]: Yes, this repository already uses Prettier.

File: packages/remark-config/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    // warn when table pipes are not aligned
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-table-pipe-alignment
+    ['lint-table-pipe-alignment', false],
+    // warn when table cells are incorrectly padded
+    // @see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-table-cell-padding
+    ['lint-table-cell-padding', 'consistent'],
+    ['lint-no-undefined-references', { allow: ['x'] }],
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/remark-config/index.js:26]: It is intended as a checkmark for the [task lists](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#task-lists): `[x]`

- [x] task
- [x] task
- [ ] task

File: packages/remark-config/index.js
Comment (literat) [reply packages/remark-config/index.js:1]: Thanks :-)

**Pull request merged**